etna/lib/etna/clients/magma/workflows/ensure_containing_record_workflow.rb

```diff
@@ -0,0 +1,44 @@
+require 'ostruct'
+require_relative './crud_workflow'
+
+module Etna
+  module Clients
+    class Magma
+      class EnsureContainingRecordWorkflow < Struct.new(:magma_crud, :models,  keyword_init: true)
```

Comment:
Might be useful for @graft .  Extracts out the logic of creating parent records to enable deep model creation from identifiers.